### PR TITLE
verify contribution from hedinasr

### PR DESCRIPTION
**Pull request by erikzaadi**
Verify CI for https://github.com/port-labs/terraform-provider-port-labs/pull/224

- Update schema to support the new parameter: pattern_jq_query
- Update documentation with examples of both usage patterns
- Add comprehensive tests for all use cases
- Adjust existing pattern validation to prevent conflicts

Description
What - Added support for dynamic pattern validation using JQ queries for string properties. The pattern_jq_query parameter allows users to specify validation rules that change based on context.

Why - This enhancement enables more flexible validation rules that can adapt to different situations without requiring multiple actions or complex conditional logic

How - Implemented by:
Adding the pattern_jq_query field to the schema with conflict validation against pattern
Modifying the API request handling to wrap JQ queries in a proper format
Adding comprehensive tests for both usage patterns and edge cases
Creating clear documentation and examples showing both approaches

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

### CODE REVIEW

**Comment by erikzaadi** [comment]
Hey @hedinasr , thanks for the contribution.
Due to security limitations, our tests only run when not on forks, I mirrored your PR here https://github.com/port-labs/terraform-provider-port-labs/pull/225 .
The tests are currently failing.

File: examples/resources/port_action/pattern_jq_query/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+```hcl
+pattern_jq_query = "[\"value1\", \"value2\", \"value3\"]"
+```
+
+This simpler format can be used to specify a fixed list of allowed values directly.
+
+## Important Notes
+
+- You cannot use both `pattern` and `pattern_jq_query` at the same time on the same property
+- The JQ query is evaluated at runtime by the Port API
```
Comment (itamar-smirra-port) [inline examples/resources/port_action/pattern_jq_query/README.md:36]:
```suggestion
- The JQ query is evaluated at runtime by Port
```

File: internal/cli/models.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ type (
 		Minimum            *float64            `json:"minimum,omitempty"`
 		Description        *string             `json:"description,omitempty"`
 		Blueprint          *string             `json:"blueprint,omitempty"`
-		Pattern            *string             `json:"pattern,omitempty"`
+		Pattern            any                 `json:"pattern,omitempty"`
```
Comment (itamar-smirra-port) [inline internal/cli/models.go:91]: Why any?

File: internal/cli/models.go [hunk 3]
```diff
@@ -88,7 +88,7 @@ type (
...
 		Enum               any                 `json:"enum,omitempty"`
```
Comment (itamar-smirra-port) [inline internal/cli/models.go:92]: We don't need to add here `PatternJqQuery` also?

File: port/action/string.go [hunk 4]
```diff
@@ -149,7 +156,6 @@ func addStringPropertiesToResource(ctx context.Context, v *cli.ActionProperty) *
 	stringProp := &StringPropModel{
 		MinLength:  flex.GoInt64ToFramework(v.MinLength),
 		MaxLength:  flex.GoInt64ToFramework(v.MaxLength),
-		Pattern:    flex.GoStringToFramework(v.Pattern),
```
Comment (itamar-smirra-port) [inline port/action/string.go:152]: What about regular patterns?

File: port/action/resource_test.go [hunk 5]
```diff
@@ -994,6 +994,290 @@ func TestAccPortActionEnum(t *testing.T) {
...
+		kafka_method = {}
+	}`, actionIdentifier)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: acctest.TestAccProtoV6ProviderFactories,
+
+		Steps: []resource.TestStep{
+			{
+				ExpectNonEmptyPlan: true,
```
Comment (itamar-smirra-port) [inline port/action/resource_test.go:1176]: We allow empty JQ queries?

File: internal/cli/models.go (hunk 3 above)
Comment (MatanPort) [reply internal/cli/models.go:92]: It's being mapped to a single entity, see previous comment

File: port/action/resource_test.go (hunk 5 above)
Comment (MatanPort) [reply port/action/resource_test.go:1176]: In TF for sure we're allowing it, I'll adjust it to validate both nil or empty

**Review by itamar-smirra-port — APPROVED** [review]
